Added a bunch of governance documents relevant to the PSO application

rbeyer opened the pull request.

rbeyer committed:
Added a bunch of documents relevant to the Planetary Software
Organization application.

michaelaye commented on Contributing.md:
```diff
@@ -0,0 +1,120 @@
+* A **Contributor** is any individual creating or commenting on an
+	issue or pull request.
+* A **Committer** is a subset of contributors who have been given
+	write access to the repository.
```
I think we need to be more clear here. We will have more than one repository? Unless we want to write PlanetaryPy GH Org? Not sure...

rbeyer replied:
So this is the `Contributing` document for the TC repo. The planetarypy repo and any affiliated package repos will also need their own `Contributing` documents that will probably be derived from this one in some way, but perhaps altered.

michaelaye replied:
Oh, gotcha.

michaelaye commented on Charter.md:
```diff
@@ -0,0 +1,135 @@
+membership decisions.  If a TC member is unable to attend a meeting
+where a planned membership decision is being made, then their consent
+is assumed.
+
```
Do we have/need a general comment somewhere about lazy consent?

rbeyer replied:
The very last item in the document is about the Consensus model.

michaelaye commented on Charter.md:
```diff
@@ -0,0 +1,135 @@
+Collaborators and given commit-access. These individuals
+are identified by the TC and their addition as Collaborators is
+discussed during the TC meeting.
+
```
How about adding: Participation in the TC is **not** a requirement for becoming a Collaborator. Or is that self-understood by not mentioning it?

rbeyer replied:
Um, so the section this text is in is titled 'Collaborators' which is a sibling of a section (before it) called 'Technical Committee Members' and both are subsctions under Membership.  So I feel like the structure of the document pretty much makes it clear that these are two categories, and that they are different.

michaelaye replied:
well, Andrew seemed to have misunderstood it?

rbeyer replied:
Well, I think he was just making a general comment, I don't think he was reviewing the document during the meeting.

michaelaye commented on Contributing.md:
```diff
@@ -0,0 +1,120 @@
+
+# Contributions
+
+Any change to resources in this repository must be through pull
```
same as above, what `repository` is meant here? The whole org? Or the core package?

rbeyer replied:
It means what it says: 'this repository,' the TC repo.  Again, each repo needs its own `Contributing` document.  I think they could all reference this one, be copied from it, or be logically descended from these contents.

michaelaye commented on Contributing.md:
```diff
@@ -0,0 +1,120 @@
+before the PR can be merged. There is no additional "sign off"
+process for contributions to be merged. Once all issues brought by
+committers are addressed it can be merged by any committer.
+
```
Even the PR author? I though we excluded that in PSO?

rbeyer replied:
If there are two approvals, and for some reason the second approver didn't merge the PR, the PR author (or any committer) could then merge it.  This is identical from the current PSO text.

michaelaye commented on Contributing.md:
```diff
@@ -0,0 +1,120 @@
+All contributors who get a non-trivial contribution merged should
+be on-boarded in a timely manner, added as a committer, and be given
+write access to the repository.
+
```
same as above, repositories?

rbeyer replied:
Ii think my answer here is the same as above: this is for this repo, and each repo will have its own document.

michaelaye commented on PSO-Application.md:
```diff
@@ -0,0 +1,114 @@
+We recognize that there is a large gulf between where we are now,
+and something as comprehensive as Astropy, and the intent is to try
+and slowly and methodically build.
+
```
is that sentence's grammar complete and okay? Feels broken off?

rbeyer replied:
I'll admit its a little grammatically artistic.  Suggest a revision.

michaelaye replied:
I was wondering if we even need to evaluate our current status compared to astropy. Isn't it sufficient to pronounce it as our target?

rbeyer replied:
Good point.  Take a look at the new version, see if you like that better.

michaelaye commented on PSO-Application.md:
```diff
@@ -0,0 +1,114 @@
+
+* Michael Aye (@michaelaye)
+* Ross Beyer (@rbeyer)
+* Andrew Annex (@AndrewAnnex)
```
are u sure you want to store members in this document? or better link to TC members?

rbeyer replied:
This PSO-application in our repo is currently a draft.  When we're ready to submit it, we need to make a PR on the PSO TSC repo.  I think the idea is that in this document (unlike our Charter where we do link), the TC member list is explicit so that application review isn't subject to external links.

rbeyer committed:
Removed the redundant comparison to Astropy, which improved the
sentence, thanks Michael.

michaelaye commented:
I guess we can merge it or do you want a second review?

rbeyer commented:
We don't currently have a rule in place for two reviews, but the changes I'm suggesting would put that rule in place. Let's see if we can get a second review.

AndrewAnnex commented on Charter.md:
```diff
@@ -0,0 +1,135 @@
+* Contribution policy
+* GitHub repository hosting
+* Conduct guidelines
+* Maintaining the list of additional Collaborators
```
lower case collaborators?

AndrewAnnex replied:
or not as I see it this way elsewhere..

rbeyer replied:
Yeah, using 'capital C' Collaborators signifies that we mean something specific in this document rather than or in addition to the dictionary definition of collaborator.

AndrewAnnex commented on Contributing.md:
```diff
@@ -0,0 +1,120 @@
+
+# Becoming a Committer
+
+All contributors who get a non-trivial contribution merged should
```
I think we should say "if they so wish" or something to that effect, there may be situations when significant contributors may not want to participate at a higher level or situations where the TC does not want to approve or invite a contributor.

rbeyer replied:
That is very reasonable.  I guess that this mode of 'automatically' becoming a committer is a simple application of the rules, rather than requiring someone to ask.  This is very equitable, but may end up with us adding a bunch of one-time committers, or those that don't want those powers, as you point out.  Look for a new commit to this PR branch in an hour or so, and I'll update this, see if you like the changed language.

michaelaye replied:
As a Committer doesn't have obligations but only added rights, i don't understand what the danger is here for someone that doesn't want to participate at a higher level? They aren't forced to?

rbeyer replied:
There's no particular 'danger' that I can think of.  From the organization side, there is a potential for a lot of adding of committers that only make a single contribution and then are never seen again, and it just becomes a 'management' issue to keep track of this list of those with commit-access and determine who really needs that access and who doesn't at some later date.

I suspect that in practice, this difference is like splitting hairs, since the TC determines who is made a Collaborator in either case.  It is just in the PSO case, that Collaborator-ship is given solely on the basis of code contribution, and nothing else, and there are also no exceptions.  In this modified case, the TC is allowed to consider other factors when admitting a Collaborator (including the wishes of a particular contributor--if they don't want write-access).  I think this language gives a little more latitude to the TC and the committers.

michaelaye replied:
Good points, thanks.

AndrewAnnex commented on PSO-Application.md:
```diff
@@ -0,0 +1,114 @@
+
+We plan to adopt the Governance model from the Planetary Software Organization.
+We plan to use the same Code of Conduct (based on the Contributor Covenant, version 1.4)
+that the PSO uses, and a Contribution model that is also similar.
```
why are they not *exactly* the same for the contribution model? Summarize the changes here.

rbeyer replied:
Also a fine point, look for the next commit to the PR, which will address this.

AndrewAnnex reviewed (changes requested):
Mostly minor questions, but I do have a question I posted on the "becoming a committer" section that I think deserves some discussion or clarification here. Do we add all significant contributors as committers if they want to be added or not, do they go through the same process of approval (assumed, but not explicitly stated)? Also, direct code contributions aren't a requirement to join as a committer in my view, only interest and good intentions. I think we should state something broader saying that anyone the TC as a whole believes should have committer access should receive it, contributions can serve as positive signals but are not the sole determining factor.

rbeyer committed:
Changed the model for a contributor becoming a Collaborator (with repo
write-access).  Rather than just automatic conversion, the TC just needs
to discuss.  Many factors could be in play beyond just code
contribution (including a contributor *not* wanting to become a
Collaborator automatically), and the TC should evaluate that spectrum.

Also enumerated this change from the PSO policy in the draft
application.

AndrewAnnex commented:
Sounds good to me, away from my computer atm but can approve changes later 

-Andrew Annex

> On Nov 27, 2019, at 2:23 PM, K.-Michael Aye <notifications@github.com> wrote:
> 
> ﻿
> @michaelaye commented on this pull request.
> 
> In Contributing.md:
> 
> > +how to get it merged or if it should merge, then it should be escalated
> +to the TC. TC members should regularly discuss pending contributions
> +in order to find a resolution. It is expected that only a small
> +minority of issues be brought to the TC for resolution and that
> +discussion and compromise among committers be the default resolution
> +mechanism.
> +
> +Exceptions to the above are minor typo fixes or cosmetic changes
> +that don't alter the meaning of a document.  Those edits can be
> +made via a PR which only requires a single approval (not two, and
> +not open for 24 h) to be merged.
> +
> +
> +# Becoming a Committer
> +
> +All contributors who get a non-trivial contribution merged should
> Good points, thanks.
> 
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub, or unsubscribe.

AndrewAnnex commented:
@rbeyer should I merge this?

rbeyer commented:
Yes.

So at this point, anyone could hit the 'merge' button (including me), since it has two approvals.
